docs/develop-guideline.md

```diff
@@ -0,0 +1,120 @@
+# 1. Overview
+
+The `enclave-agent` runs on a SGX-based LibOS. In the runtime, the memory and the mounted filesystem used by the enclave agent is encrypted by the SGX. 
+
+The supporting LibOS includes
+- [occlum](https://github.com/occlum)
+    - the encrypted filesystem is called by `unionfs`
+- [gramine](https://github.com/gramineproject/gramine)
+
+The dependencies installation process can be found in the [develop-env-install.md](develop-env-install.md).
+
+
+## Image Management
+
+The enclave agent plays the role of the container image management. It is responsible for four basic missions, including
+- Pulling: it pulls the index, manifest, config and layers of a image, which is dependent on the crate [image-rs](https://github.com/confidential-containers/image-rs)
+- Unpacking: it decrypts and decompresses the layers of a image if necessary. The decryption is dependent on
+the crate [ocicrypt-rs](https://github.com/confidential-containers/ocicrypt-rs).
+- Storing: it saves the unpacking layers on an encrypted filesystem
+- Mounting: it merges layers to a rootfs for a application container and mounts the rootfs to a mounting point
+
+
+## Security Management
+
+The enclave agent need to satisfy security requirements, including
+- attestation
+    - remote attestation: it provides security information for the remove 
+    - local attestation
+    - the crate [attestation-agent](https://github.com/confidential-containers/attestation-agent) supports a native attesation client
+- signature verification: 
+    - some images are signed by the pulisher or the store or both, it need to verify these signatures
+    - the crate [attestation-agent](https://github.com/confidential-containers/attestation-agent) supports a native signature client 
+
+# 2. Development Guide
```

Comment:
The tools added in this PR are already available in this repository so we don't need to add them again. Please check how this flow works for you and then we can update the docs here:

```
$ SGX_MODE=SIM PAYLOAD_ARTIFACTS=/tmp/payload ./gen_bundle.sh foo:test
$ cd /tmp/payload
$ mkdir rootfs
$ sudo tar xf agent-instance.tar -C rootfs/
$ runc run 123
```

Reply:
The `tools/build/agent-enclave-build/gen_bundle.sh` and `test_files/gen_bundles.sh` have a similar workflow, but create different bundles, The differences are
- `agent-enclave-build/gen_bundle.sh` installs all dependencies such as occlum in the bundle and the container can be deployed on any related platform and be launched by `runc`. 
- `test_files/gen_bundles.sh` does not install dependencies. The dependencies are installed on the development platform before. The bundle only can be launched by `rune`. And all intermediate products of the bundle building can be found on the development platform. The script is aimed at coding, building and debugging the enclave agent as quickly as possible.

It consumes too much time to build an entire bundle for debugging.  Maybe we can rename the `test_files/gen_bundles.sh` to `test_files/gen_bundles_debug.sh`

Reply:
please add suggestions to #45 to have the payload image creation better suitable for development needs.

> It consumes too much time to build an entire bundle for debugging. 

currently it takes time because the first thing the script does is it deletes the previously built artifacts. One immediate optimization is to make sure developers can the bundle build gets to reuse cached layers as much as possible.